### ISSUE TITLE
[ZTP] sonic-ztp service runs every boot and incurs high CPU usage during boot even when disabled

### ISSUE THREAD

**Issue by stepanblyschak** [sonic-net/sonic-buildimage#20741]
#### Description

sonic-ztp service runs every boot and incurs high CPU usage during boot.

#### Steps to reproduce the issue:
1. Boot the system, ZTP is disabled
2. Measure CPUUsageNSec of ztp service

#### Describe the results you received:

High CPU usage of ztp service during boot:
```
root@sonic:/home/admin# systemctl show ztp --property=CPUUsageNSec
CPUUsageNSec=3665570915
```

#### Describe the results you expected:

ztp should not run if disabled and should not add CPU load when feature is not used.

#### Output of `show version`:

```
SONiC Software Version: SONiC.202405_RC.44-28a64576c_Internal
SONiC OS Version: 12
Distribution: Debian 12.7
Kernel: 6.1.0-22-2-amd64
Build commit: 28a64576c
Build date: Wed Nov  6 16:40:57 UTC 2024
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by prgeor** [comment]
@adyeung can you check if this is a day-1 issue on ZTP

**Comment by adyeung** [comment]
@rajendra-dendukuri please help take a look

**Comment by rajendra-dendukuri** [comment]
Agreed. This optimization can be made to not spin the cpu cycles when ztp is indeed administratively disabled. I proposed to disable and mask ztp service when the ztp mode is disabled. Conversely, when ztp is enabled, the service is unmasked and enabled. 

Please let me if this is acceptable.

https://gist.github.com/rajendra-dendukuri/4d8a72ee38c73e6479434fb7dc23e79e

**Comment by stepanblyschak** [comment]
@rajendra-dendukuri Thanks. One question - when installing or upgrading to new image is ztp service disabled & masked by default or could it be controlled by some build option?

**Comment by rajendra-dendukuri** [comment]
> @rajendra-dendukuri Thanks. One question - when installing or upgrading to new image is ztp service disabled & masked by default or could it be controlled by some build option?

As rightly pointed there are a few scenarios we need address.

1. ZTP is admin disabled on a system, operator installs a new image which has ztp enabled inside the image
>> When the new image boots, ztp service should not start and be masked. Here, on first boot we may have to disable and mask the ztp service. We can do this in rc.local or config-setup service.

2. ZTP is admin disabled inside an image. User installs this image on a system with ztp enabled
>> When the new image boots, ztp service will remain unmasked and enabled. Here, on first boot we may have to enable and unmask the ztp service. We can do this in rc.local or config-setup service.

3. ZTP is admin disabled inside an image. User installs this image on a system with ztp disabled or is installed from ONIE
>> When the image boots, ztp service will be masked and disabled. Here, on first boot we may have to disable and mask the ztp service. We can do this in rc.local or config-setup service.

The default ztp admin mode is determined in the file "src/usr/lib/python3/dist-packages/ztp/defaults.py" with key "admin-mode"       . This is a build time modification. This is what I read as "ztp is included and enabled in the image".

Since there are multiple combinations, the original design didn't do this to avoid situations where you install an image using ztp and expect the newly install image to continue and finish off ztp. 

We can discuss.
